pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/graphql_helpers_test.go

```diff
@@ -144,17 +144,20 @@ func graphqlMockServer(responses *responses) *http.ServeMux {
 func restMockServer(resp responses) *http.ServeMux {
 	var mux http.ServeMux
 	mux.HandleFunc("/api-v3/repos/o/r/contributors", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(resp.responseCode)
 		if resp.responseCode == http.StatusOK {
-			contribs, err := json.Marshal(resp.contribs)
+			contribs, err := json.Marshal(resp.contribs[resp.page])
 			if err != nil {
 				fmt.Printf("error marshalling response: %v", err)
 			}
-			// Attempt to write data to the response writer.
+			//The link header is used to paginate through the results and the page refers to pages left, so we need to decrement.
+			//An example is with 2 expected responses, the first page will have a value of 1 (2-0-1), then on the second it will be 0 (2-1-1)
+			link := fmt.Sprintf("<https://api.github.com/repositories/39840932/contributors?per_page=100&page=%d>; rel=\"next\"", len(resp.contribs)-resp.page-1)
```

Comment:
With the link header, there is still no call to api.github.com though correct?

Reply:
Nope, just took that from a real api call to see exactly what is meant to get passed back